### PR TITLE
Disallow clearing timers without no argument at all

### PR DESCRIPTION
**Pull request by Andarist**
This aligns the signatures with the recent change in `@types/node` ([here](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/59914)).

My rationale behind this change is outlined [here](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/59914#issuecomment-1122085478), but in a short, I don't think those calls make any sense:
```js
clearTimeout() // what do we clear?
clearInterval() // what do we clear?
```

### CODE REVIEW

File: inputfiles/overridingTypes.jsonc [hunk 1]
```diff
@@ -218,6 +218,22 @@
                         }
                     }
                 }
+            },
+            "WindowOrWorkerGlobalScope": {
+                "methods": {
+                    "method": {
+                        "clearInterval": {
+                            "overrideSignatures": [
+                                "clearInterval(id: number | undefined): void"
```
Comment (saschanaz) [inline inputfiles/overridingTypes.jsonc:227]: `overrideSignatures` should be used as a last resort. Could you do the `signature` way as it's done below in OnErrorEventHandlerNonNull etc.?

File: inputfiles/overridingTypes.jsonc [hunk 2]
```diff
@@ -218,6 +218,38 @@
...
+                            }
+                        },
+                        "clearTimeout": {
+                            "signature": {
+                                "0": {
+                                    "param": [
+                                        {
+                                            "name": "id",
+                                            "optional": "false",
+                                            "overrideType": "number | undefined"
```
Comment (saschanaz) [inline inputfiles/overridingTypes.jsonc:245]: `overrideType` should be also a last resort, perhaps `additionalTypes: ["undefined"]`?

**Comment by Andarist** [comment]
@saschanaz thanks for the suggestions. Sorry for a few rapid pushes - I forgot that I had to accept baselines on top of rerunning the build script and didn't know at first how I should verify my changes locally 😅 I think this build will finally pass and it comes with your suggestions applied.

**Comment by saschanaz** [comment]
Thanks!

lgtm

**Pull request merged**